### ISSUE TITLE
after training for days on GPU...

### ISSUE THREAD

**Issue by darenr** [llSourcell/tensorflow_chatbot#29]
...the conversations are gibberish (

```[me] hello
[tf] Forget here , look .
[me] are you a robot
[tf] Oh not .
[me] what is your name?
[tf] Long it is always Isn ' d just drop with for awhile
```
tensorflow 0.12

training:

```global step 513600 learning rate 0.0100 step-time 0.76 perplexity 1.01
  eval: bucket 0 perplexity 1153481202401.01
  eval: bucket 1 perplexity 3621452378204.35
  eval: bucket 2 perplexity 5702366989115.59
  eval: bucket 3 perplexity 294104169518.42
```

settings in ini (python 2.7, GFX GPU, Linux)

```[ints]
# vocabulary size
enc_vocab_size = 40000
dec_vocab_size = 40000
num_layers = 3
layer_size = 256
max_train_data_size = 0
batch_size = 64
steps_per_checkpoint = 300
[floats]
learning_rate = 0.5
learning_rate_decay_factor = 0.99
max_gradient_norm = 5.0
```

so I'm wondering if I should look elsewhere for different conversational data? Is there a tool in your repo for turning a dataset into the enc/dec format you're expecting? I can write one but didn't want to duplicate work.

**Comment by Graystripe17** [comment]
That's disappointing.

Hey I don't wanna impose, but do you think you could create a branch or fork where you post the whole repo and it's learning data? I know it's kind of cheating but my computer gets super hot trying to compute.

**Comment by nabihach** [comment]
Thanks so much for this post. Very helpful.

**Comment by jmoney4769** [comment]
Try a significantly smaller learning rate (play with it but you could start with 2-3 0's).  The gradients you compute with that learning rate are so large that Gradient Descent will overshoot every time and continuously make things worse.

**Comment by isaacdchan** [comment]
@darenr out of curiosity, what GPU were you using where it took days?
